test/distributed/tensor/parallel/test_tp_examples.py

```diff
@@ -94,11 +94,16 @@ def _test_mlp_training_e2e(self, is_seq_parallel=False, recompute_activation=Fal
         optim_tp = torch.optim.SGD(model_tp.parameters(), lr=LR)
 
         output = model(inp)
-        output_tp = model_tp(inp)
-        self.assertEqual(output, output_tp)
-
         output.sum().backward()
-        output_tp.sum().backward()
+
+        from torch.distributed._tensor.debug import CommDebugMode
+        comm_mode = CommDebugMode()
+        with comm_mode:
+            output_tp = model_tp(inp)
+            output_tp.sum().backward()
+
+        self.assertEqual(output, output_tp)
+        self.assertEqual(comm_mode.get_comm_counts()[c10d_functional.all_reduce], 1)
```

Comment:
Seems the test is complaining about this check for `test_mlp_training`.

Reply:
yeah adjusted